Source/UnitTests/Common/BitUtilsTest.cpp

```diff
@@ -88,3 +90,50 @@ TEST(BitUtils, IsValidLowMask)
   EXPECT_FALSE(Common::IsValidLowMask((u64) ~(0b10000)));
   EXPECT_FALSE(Common::IsValidLowMask((u64)(~((u64)(~0b0) >> 1) | 0b1111)));
 }
+
+TEST(BitUtils, BitCastPtr)
+{
+  std::vector<u8> data{0, 1, 0, 2, 0, 3, 0, 0xFF, 0xFF};
+  u8* buffer = data.data();
+
+  EXPECT_EQ(s16(1), Common::BitCastPtr<s16>(buffer + 1));
```

Comment:
A few of these tests assume a little-endian host. I guess that's ultimately fine, since we're not anywhere close to being able to target big-endian hosts, but maybe at least add a comment?

Reply:
I added an additional test checking the endianness of the host.